app/src/main/java/com/conupods/OutdoorMaps/BuildingDataMap.java

```diff
@@ -0,0 +1,84 @@
+package com.conupods.OutdoorMaps;
+
+import android.content.res.AssetManager;
+
+import com.conupods.App;
+import com.google.android.gms.maps.model.LatLng;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+
+/**
+ * A singleton class that holds hardcoded building data of concordia buildings.
+ * The class parses the data (only once) that is saved in the file ../assets/buildings.json
+ * who was imported from the concordia API manually.
+ */
+public class BuildingDataMap {
+    private static BuildingDataMap mInstance;
+    private HashMap<LatLng, Building> mData;
+
+    private BuildingDataMap() throws Exception {
+        // todo possibly change the index to be the marker object and not latlng
+        mData = new HashMap<>();
+        parseBuildingData();
+    }
+
+    public static synchronized BuildingDataMap getInstance() {
+        if (mInstance == null) {
+            try {
+                mInstance = new BuildingDataMap();
+            } catch (Exception e) {
+                e.printStackTrace();
+                return null;
+            }
+        }
+        return mInstance;
+    }
```

Comment:
Nice job on the singleton!

Reply:
Can we move this class to the services package? Mike and I need exactly this for the search feature.